pygmt/src/pygmtlogo.py

```diff
@@ -0,0 +1,258 @@
+    shape : str
+        Shape of the visual. Use ``"circle"`` for a circle shape [Default] or
+        ``"hexagon"`` for a hexagon shape.
+    wordmark : bool, str
+        Add the wordmark "PyGMT" and adjust its orientation relative to the visual.
+        Set to ``True`` or ``"horizontal"``, to add the wordmark at the right side of
+        the visual [Default]. Use ``"vertical"`` to place the wordmark below the
+        visual and ``False`` to add no wordmark.
+    """
+
+    # Helpful definitions
+    size = 4
+    region = [-size, size] * 2
+    projection = "x1c"
+
+    # Outer and inner radii of compass lines
+    r1, r2 = size * 0.625, size * 0.325
+
+    # Rotation around z (vertical) axis placed in the center
+    # Has to be applied to each plotting command, up on second call set to True
+    perspective = "30+w0/0"  # Rotation by 30 degrees
+
+    # Define colors
+    color_light = "white"
+    color_dark = "gray20"
+
+    # visual
+    color_blue = "48/105/152"  # Python blue
+    color_yellow = "255/212/59"  # Python yellow
+    color_red = "238/86/52"  # GMT red
+    if not color:
+        color_blue = color_yellow = color_red = color_dark
+        if theme == "dark":
+            color_blue = color_yellow = color_red = color_light
+
+    # background and wordmark
+    match theme:
+        case "light":
+            color_bg = color_light
+            color_py = color_blue
+            color_gmt = color_dark
+        case "dark":
+            color_bg = color_dark
+            color_py = color_yellow
+            color_gmt = color_light
+
+    # Define shape
+    match shape:
+        case "circle":
+            symbol = "c"  # circle
+            diameter = 7.5
+            diameter_add = 0.5
+        case "hexagon":
+            symbol = "h"  # hexagon
+            diameter = 8.6
+            diameter_add = 0.6
+
+    # Define wordmark
+    font = "AvantGarde-Book"
+    match wordmark:
+        case "vertical":
+            args_text_wm = {"x": 0, "y": -5, "justify": "CT", "font": f"2.5c,{font}"}
+        case True | "horizontal":
+            args_text_wm = {"x": 6, "y": 0, "justify": "LM", "font": f"8c,{font}"}
+
+    fig = pygmt.Figure()
+
+    # blue circle / hexagon for Earth
+    fig.plot(
+        x=0,
+        y=0,
+        region=region,
+        projection=projection,
+        style=f"{symbol}{diameter}c",
+        pen=f"0.5c,{color_blue}",
+        fill=color_bg,
+        perspective=perspective,
+        no_clip=True,  # needed for corners of hexagon shape
+    )
+
+    # yellow lines for compass
+    lines_yellow = [
+        ([-size, size], [0, 0]),  # horizontal line
+        ([-r1, -r2], [r1, r2]),  # upper left
+        ([-r1, -r2], [-r1, -r2]),  # lower left
+        ([r1, r2], [r1, r2]),  # upper right
+        ([r1, r2], [-r1, -r2]),  # lower right
+    ]
+    for x, y in lines_yellow:
+        fig.plot(x=x, y=y, pen=f"5p,{color_yellow}", perspective=True)
```

Comment:
Draw three-segment horiontal lines:
```suggestion
    # Compass
    x1, x2 = r1 * 0.7071, r2 * 0.7071  # sqrt(2)/2 = 0.7071
    lines_compass = [ 
        ([-r0, -r2], [0, 0]),  # horizontal lines
        ([-r4, 0], [0, 0]), 
        ([r2, r0], [0, 0]),
        ([-x1, -x2], [x1, x2]),  # upper left
        ([-x1, -x2], [-x1, -x2]),  # lower left
        ([x1, x2], [x1, x2]),  # upper right
        ([x1, x2], [-x1, -x2]),  # lower right
    ] 
    for x, y in lines_compass:
        fig.plot(x=x, y=y, pen=f"5p,{color_yellow}", perspective=True)
```